File: ga/glm4_result_2024_6/result/1968C_cpp_glm/17_1968C.cpp
const N: usize = 500;
const X: usize = 500;
const A: i32 = 1000000000;

fn proc() {
    let mut n = 0;
    let mut x = [0; N];
    let mut a = [0; N];

    // Read the number of elements
    std::io::stdin().read_line(&mut String::new()).unwrap();
    std::io::stdin().read_line(&mut String::new()).unwrap();
    n = std::io::stdin().read_line(&mut String::new()).unwrap().trim().parse::<usize>().unwrap();

    // Read the array elements
    for i in 1..n {
        std::io::stdin().read_line(&mut String::new()).unwrap();
        x[i] = std::io::stdin().read_line(&mut String::new()).unwrap().trim().parse::<i32>().unwrap();
    }

    // Initialize the first element of a
    a[0] = x[1] + 1;

    // Compute the array a based on x
    for i in 1..n {
        let mut j = 0;
        while j <= A as usize {
            a[i] = x[i] + j as i32 * a[i - 1];
            if (i + 1 < n) && (x[i + 1] < a[i]) && (a[i] != a[i - 1]) {
                break;
            }
            j += 1;
        }
    }

    // Print the array a
    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let mut t = 0;
    std::io::stdin().read_line(&mut String::new()).unwrap();
    t = std::io::stdin().read_line(&mut String::new()).unwrap().trim().parse::<i32>().unwrap();

    while t > 0 {
        proc();
        t -= 1;
    }
}

File: ga/4omini_result_2024_7/result/1968C_cpp_mini/17_1968C.rs
use std::io::{self, BufRead};

const N: usize = 500;
const A: i64 = 1_000_000_000;

fn proc() {
    let stdin = io::stdin();
    let mut input = stdin.lock().lines();

    let n: usize = input.next().unwrap().unwrap().parse().unwrap();
    let mut x: Vec<i64> = vec![0; N];
    let mut a: Vec<i64> = vec![0; N];

    for i in 1..n {
        x[i] = input.next().unwrap().unwrap().parse().unwrap();
    }

    a[0] = x[1] + 1;

    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j * a[i - 1];

            if (i + 1 < n).then(|| x[i + 1])   // Condition to check for next element in the array `x`
                .unwrap_or(1) < a[i] && a[i] != a[i - 1] {
                break;
            }
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let stdin = io::stdin();
    let input = stdin.lock().lines().next().unwrap().unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        proc();
    }
}